proto/substrait/plan.proto

```diff
@@ -46,6 +46,11 @@ message Plan {
   // unused. In many cases, a consumer may be able to work with a plan even if
   // one or more message types defined here are unknown.
   repeated string expected_type_urls = 5;
+
+  // An optional list of bindings for dynamic parameters used in this plan.
+  // The index of each entry in the list corresponds to the parameter_anchor
+  // of a DynamicParameter expression, mapping it to its runtime value.
+  repeated Expression.Literal parameter_bindings = 7;
```

Comment:
My inclination is to match the SimpleExtensionURI pattern here. In that pattern, the anchors are in a message and this repeated element doesn't necessarily have to be in order. This means numbers are more explicit and we have a message to add other properties in the future. I regret using the implicit pattern in ReferenceRel and think we should probably avoid further impls of that pattern. Thoughts?

Reply:
I'm okay with that but it is half the work of names without the strings.  You will have to index the entries here so you can refer to them later just like you would need to do with strings.  If we do take that route we could consider using a map here so that at least the indexing would be done by the proto library instead of everyone needing to roll their own.

Reply:
One option is to use the SimpleExtensionURI pattern since that's what is already in place and existing libraries can leverage the same existing deserialization logic built for it (that's what I did in this PR originally). If we want to handle things differently (i.e., dictionary-based), it could be worth having a broader discussion, including migrating other existing fields? That way, we avoid introducing yet another approach (i.e., explicit integer anchor, implicit list positional anchor, and now a dictionary-based anchor). That said, I'm fine with either approach, let me know and I'll adapt the PR accordingly.

Reply:
I'm inclined to match SimpleExtensionURI. I don't think we should use map here. Part of my sensitivity is that in many packages we do things like rewrites of these structures. Having the ID outside the structure is frustrating. The same thing happens in rest apis. Some people are inclined to say "hey, the url has the id, we don't need it in the object". The reality is that having it in the object fits most languages much better since odds are that is how they will be manipulated.

Reply:
OK, I have updated the code accordingly.